From 65d889651d3f45913f3dde2f6f055a887e673512 Mon Sep 17 00:00:00 2001
From: Nick Liu <nick_liu@avalue.com.tw>
Date: Wed, 1 Jun 2016 11:55:19 +0800
Subject: [PATCH 1/1] Fixed lvds.

modified:   kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts
modified:   kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi
modified:   kernel/arch/arm/configs/imx_v7_defconfig
---
 .../arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts  |  4 +-
 kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi        | 67 ++++++++++++++--------
 kernel/arch/arm/configs/imx_v7_defconfig           |  2 +-
 3 files changed, 46 insertions(+), 27 deletions(-)

diff --git a/kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts b/kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts
index 5a801eb..4c80d8c 100644
--- a/kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts
+++ b/kernel/arch/arm/boot/dts/imx6q-rev-sa01-pfuze100.dts
@@ -22,11 +22,11 @@
 
 &ldb {
 	lvds-channel@0 {
-		crtc = "ipu2-di0";
+		crtc = "ipu2-di1";
 	};
 
 	lvds-channel@1 {
-		crtc = "ipu2-di1";
+		crtc = "ipu2-di0";
 	};
 };
 
diff --git a/kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi b/kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi
index f07f3f4..0a20905 100644
--- a/kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi
+++ b/kernel/arch/arm/boot/dts/imx6qdl-smarc.dtsi
@@ -91,6 +91,18 @@
 			gpio = <&gpio6 14 0>;
 			enable-active-high;
 		};
+
+        reg_lvds_en: lvds_en {
+            compatible = "regulator-fixed";
+            regulator-name = "lvds_en";
+            gpio = <&gpio2 22 GPIO_ACTIVE_HIGH>;
+        };
+
+        reg_lvds_bkl_en: lvds_bkl_en {
+            compatible = "regulator-fixed";
+            regulator-name = "lvds_bkl_en";
+            gpio = <&gpio3 00 GPIO_ACTIVE_HIGH>;
+        };
 	};
 
 	sound {
@@ -245,7 +257,7 @@
 
 &dcic2 {
 	dcic_id = <1>;
-	dcic_mux = "dcic-lvds1";
+	dcic_mux = "dcic-lvds0";
 	status = "okay";
 };
 
@@ -572,6 +584,13 @@
 			>;
 		};
 
+        pinctrl_lvds: lvdsengrp {
+            fsl,pins = <
+                MX6QDL_PAD_EIM_A16__GPIO2_IO22      0x80000000
+                MX6QDL_PAD_EIM_DA0__GPIO3_IO00      0x80000000
+            >;
+        };
+
         pinctrl_uart1: uart1grp {
 			fsl,pins = <
 				MX6QDL_PAD_CSI0_DAT10__UART1_TX_DATA	0x1b0b1
@@ -651,20 +670,21 @@
 	lvds-channel@0 {
 		fsl,data-mapping = "spwg";
 		fsl,data-width = <18>;
-		status = "okay";
+        primary;
+        status = "okay";
 
 		display-timings {
 			native-mode = <&timing0>;
 			timing0: hsd100pxn1 {
-				clock-frequency = <65000000>;
-				hactive = <1024>;
-				vactive = <768>;
-				hback-porch = <220>;
-				hfront-porch = <40>;
-				vback-porch = <21>;
-				vfront-porch = <7>;
-				hsync-len = <60>;
-				vsync-len = <10>;
+                clock-frequency = <25000000>;
+                hactive = <640>;
+                vactive = <480>;
+                hback-porch = <1 114 139>;
+                hfront-porch = <1 16 116>;
+                vback-porch = <30 32 34>;
+                vfront-porch = <1 10 45>;
+                hsync-len = <5 30 5>;
+                vsync-len = <1 3 5>;
 			};
 		};
 	};
@@ -672,22 +692,21 @@
 	lvds-channel@1 {
 		fsl,data-mapping = "spwg";
 		fsl,data-width = <18>;
-		primary;
-		status = "okay";
+        status = "okay";
 
 		display-timings {
 			native-mode = <&timing1>;
 			timing1: hsd100pxn1 {
-				clock-frequency = <65000000>;
-				hactive = <1024>;
-				vactive = <768>;
-				hback-porch = <220>;
-				hfront-porch = <40>;
-				vback-porch = <21>;
-				vfront-porch = <7>;
-				hsync-len = <60>;
-				vsync-len = <10>;
-			};
+                clock-frequency = <25000000>;
+                hactive = <640>;
+                vactive = <480>;
+                hback-porch = <1 114 139>;
+                hfront-porch = <1 16 116>;
+                vback-porch = <30 32 34>;
+                vfront-porch = <1 10 45>;
+                hsync-len = <5 30 5>;
+                vsync-len = <1 3 5>;
+            };
 		};
 	};
 };
@@ -719,7 +738,7 @@
 
 &pwm1 {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_pwm1>;
+    pinctrl-0 = <&pinctrl_pwm1 &pinctrl_lvds>;
 	status = "okay";
 };
 
diff --git a/kernel/arch/arm/configs/imx_v7_defconfig b/kernel/arch/arm/configs/imx_v7_defconfig
index f9c51c1..bbbef9b 100644
--- a/kernel/arch/arm/configs/imx_v7_defconfig
+++ b/kernel/arch/arm/configs/imx_v7_defconfig
@@ -2825,7 +2825,7 @@ CONFIG_BACKLIGHT_PWM=y
 # CONFIG_BACKLIGHT_LM3630A is not set
 # CONFIG_BACKLIGHT_LM3639 is not set
 # CONFIG_BACKLIGHT_LP855X is not set
-# CONFIG_BACKLIGHT_GPIO is not set
+CONFIG_BACKLIGHT_GPIO=y
 # CONFIG_BACKLIGHT_LV5207LP is not set
 # CONFIG_BACKLIGHT_BD6107 is not set
 # CONFIG_VGASTATE is not set
-- 
1.9.1

